datasets/mx/governors/crawler.py

```diff
@@ -0,0 +1,124 @@
+from zavod.logic.pep import categorise
+
+def extract_birth_place_and_date(link_governor_page, context: Context):
+    """
+    The birth place and date can be found in the page of each governor
+    in a ul tag. It is not always in the same position in the list,
+    so we will iterate over all elements and find the one that has
+    the string "Lugar y fecha de nacimiento:". If it is not found,
+    we raise a warning and return None.
+
+    :param link_governor_page: Link to a page of a specific governor
+    :param context: The context object.
+    """
+
+    response = context.fetch_html(link_governor_page)
+
+    path_to_information = '//*[@id="fContenido"]//li'
+
+    for element in response.xpath(path_to_information):
+        text = ' '.join(element.itertext())
+        if 'Lugar y fecha de nacimiento:' in text:
+            pattern = r'Lugar y fecha de nacimiento: ([^.]+\.) (\d{1,2}/\d{1,2}/\d{4})\.'
+            match = re.search(pattern, text)
+            if match:
+                birth_place = match.group(1)
+                birth_date = match.group(2)
+                return birth_place, birth_date
+
+    context.log.warning("Failed to identify birth place and date, link: {}".format(link_governor_page))
+    return None, None
+
+
+def crawl_item(input_html, context: Context):
+    """
+    Creates an entity, a position and a occupancy from the raw HTMLElement.
+
+    :param input_html: The div representing that governor
+    :param context: The context object.
+    """
+
+    link_governor_page = input_html.xpath("./div/div/div/div[1]/a/@href")[0]
+
+    # The easiest way to get the data from the HTML is surprisingly from the text of the class
+    # All the information in the web site is written there in the form:
+    # col-xs-12 col-sm-6 col-md-4 pb15 mix {name of the governor} [{state}] {start_date} {end_date}
+    # We will then use a regex to extract the name, state, start_date and end_date
+
+    regex_pattern = r'^col-xs-12 col-sm-6 col-md-4 pb15 mix (.*?) \[(.*?)\] (\d{2}/\d{2}/\d{4}) (\d{2}/\d{2}/\d{4})$'
+
+    match = re.match(regex_pattern, input_html.get('class'))
+
+    if match is None:
+        context.log.warning("Unable to extract information from HTML element")
+
+    name, state, start_date, end_date = match.group(1), match.group(2), match.group(3), match.group(4)
+
+    person = context.make("Person")
+    person.id = context.make_id(name)
+
+    # The names in the card always have a title as the first word, for example,
+    # Ing. Carlos Lozano de la Torre, where Ing. stands for Engineer
+    # We will add two name propreties, one with the "full" name
+    # and other with the clean name. The clean name
+    # can be retrived from the url to that governors page
+    person.add("name", name)
+    clean_name = (link_governor_page
+                  .replace("https://www.conago.org.mx/miembros/detalle/", "")
+                  .replace("-", " ")
+                  .title())
+    person.add("name", clean_name)
```

Comment:
Sorry, only noticed this now.
Instead of de-slugifying the name, I think a better approach is

1. take the name from the h4 in the card
2. strip `r"^([A-Z][a-z]*\\.)+ "` from it

Then we have the names capitalised and from the field we can expect quite safely to contain the name.

Adding it twice with and without ligatures feels more messy than the above.

Reply:
I can do like this where I can just strip the first word in the h4 tag. 

My concern is that if we only put the name that is in the title, when someone searches for the name of a governor in the open sanctions database it wouldn't match because the name property has the title, or this won't be a problem?

Reply:
So the regex I proposed should strip any abbreviated word from the start of the heading, so it should catch all the titles and nothing else. That would clean the name up and match nicely. But people with many names could well match in our API even if a title slips in. Still, I think it's easy enough to strip titles here.